Engine/system/api/x11api.cpp

```diff
@@ -188,7 +189,21 @@ SystemApi::Window *X11Api::implCreateWindow(Tempest::Window *owner, uint32_t w,
   }
 
 SystemApi::Window *X11Api::implCreateWindow(Tempest::Window *owner, SystemApi::ShowMode sm) {
-  return implCreateWindow(owner,800,600); //TODO
+  Screen* s = DefaultScreenOfDisplay(dpy);
+  int width = s->width;
+  int height = s->height;
+  
+  SystemApi::Window* hwnd = nullptr;
+
+  if(sm==Normal) 
```

Comment:
Better to do a switch-case here, for better tracking of sm:
```c++
switch(sm) {
  case Hidden: 
    hwnd = implCreateWindow(owner,1, 1);
    // TODO: make window invisible
    break;
  case Normal: 
    hwnd = implCreateWindow(owner,800, 600);
    break;
  case Minimized: 
  case Maximized: 
    // TODO
    hwnd = implCreateWindow(owner,800, 600);
    break;
  case FullScreen: 
    hwnd = implCreateWindow(owner,800, 600);
    implSetAsFullscreen(hwnd,true);
    break;
  }
``

Reply:
I've updated the code to use the switch statement as described.